modules/gateway/peers.go

```diff
@@ -257,31 +299,33 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 		},
 		sess: muxado.Client(conn),
 	})
-	g.mu.Unlock(id)
+	g.mu.Unlock()
 
 	// call initRPCs
-	id = g.mu.RLock()
+	g.mu.RLock()
 	for name, fn := range g.initRPCs {
-		go g.RPC(addr, name, fn)
+		go func(name string, fn modules.RPCFunc) {
+			g.RPC(addr, name, fn)
+		}(name, fn)
```

Comment:
interesting that this got mangled. Looks like a statement was added in the body, then removed later?